src/main/java/org/corfudb/cmdlets/corfu_layout.java

```diff
@@ -252,6 +271,49 @@
                         e.toString(),
                         "stack: " + ExceptionUtils.getStackTrace(e));
             }
+        } else if ((Boolean) opts.get("update_layout")) {
+            // This is not strictly a low-level primitive for the layout Paxos protocol.
+            // Rather, it's here at glue for QuickCheck testing of higher-level
+            // layout API exercise in the same spirit that layout_qc.erl tests
+            // the lower-level Paxos protocol.
+
+            LayoutView lv;
+            CorfuRuntime rt = configureRuntimeAddrPort(opts);
+            if ((lv = layoutViews.get(addressportPrefix + addressport)) == null) {
+                log.trace("Creating LayoutView for {} ++ {}:{}", addressportPrefix, port);
+                lv = new LayoutView(rt);
+                layoutViews.putIfAbsent(addressportPrefix + addressport, lv);
+            }
+            lv = layoutViews.get(addressportPrefix + addressport);
+
+            Layout l = getLayout(opts);
+            l.setRuntime(rt);
+            long rank = Long.parseLong((String) opts.get("--rank"));
+            log.debug("update_layout with layout={}, rank={}, ", l, rank);
+            try {
+                // Important: we must add a CorfuRuntime to the Layout object l.
+                // If we don't, then we'll get a NullPointerException deep in the
+                // guts of LayoutView.committed() which tries to use the layout's
+                // runtime member.
+                log.trace("Specify (2) router's epoch as " + clientEpoch);
+                rt.getRouter((String) opts.get("<address>:<port>")).setEpoch(clientEpoch);
+                l.setRuntime(rt);
+                lv.updateLayout(l, rank);
+                return cmdlet.ok();
+            } catch (WrongEpochException we) {
+                return cmdlet.err("Exception (1) during updateLayout",
+                        we.getCause() == null ? "WrongEpochException" : we.getCause().toString(),
+                        "correctEpoch: " + we.getCorrectEpoch(),
+                        "stack: " + ExceptionUtils.getStackTrace(we));
+            } catch (OutrankedException oe) {
+                return cmdlet.err("Exception (2) during updateLayout",
+                    oe.getCause() == null ? "OutrankedException" : oe.getCause().toString(),
+                    "newRank: " + Long.toString(oe.getNewRank()),
+                    "stack: " + ExceptionUtils.getStackTrace(oe));
+            } catch (QuorumUnreachableException ue) {
+                return cmdlet.err("Exception (2) during updateLayout",
+                        ue.getCause().toString());
+            }
         }
         return cmdlet.err("Hush, compiler.");
```

Comment:
Very informal error which needs to be changed and doesn't tell the user anything. What does it mean if I'm running my cmdlet and I get an error at runtime that says "Hush, compiler."?. (Hush compiler? Maybe you need to fix the args to cmdlet.err ...)